odl/tomo/geometry/spect_geometry.py

```diff
@@ -0,0 +1,156 @@
+        det_init_axes = (det_init_axis_0, axis)
+        detector = Flat2dDetector(dpart, det_init_axes)
+
+        self._detector = detector
+        self._motion_params = odl.Interval(apart.points()[0],
+                                           apart.points()[-1])
+        self._motion_part = apart
+
+    @property
+    def motion_partition(self):
+        """Partition of the motion parameter set into subsets."""
+        return self._motion_part
+
+    @property
+    def motion_params(self):
+        """Continuous motion parameter range, an `IntervalProd`."""
+        return self.motion_partition.set
+
+    @property
+    def motion_grid(self):
+        """Sampling grid of `motion_params`."""
+        return self.motion_partition.grid
+
+    @property
+    def detector(self):
+        """Detector representation of this geometry."""
+        return self._detector
+
+    @property
+    def det_partition(self):
+        """Partition of the detector parameter set into subsets."""
+        return self.detector.partition
+
+    @property
+    def det_params(self):
+        """Continuous detector parameter range, an `IntervalProd`."""
+        return self.detector.params
+
+    @property
+    def det_grid(self):
+        """Sampling grid of `det_params`."""
+        return self.detector.grid
+
+    @property
+    def partition(self):
+        """Joined parameter set partition for motion and detector.
+
+        Returns a `RectPartition` with the detector partition inserted
+        after the motion partition.
+        """
+        return self.det_partition.insert(0, self.motion_partition)
+
+    @property
+    def params(self):
+        """Joined parameter set for motion and detector.
+
+        By convention, the motion parameters come before the detector
+        parameters.
+        """
+        return self.partition.set
+
+    @property
+    def grid(self):
+        """Joined sampling grid for motion and detector.
+
+        By convention, the motion grid comes before the detector grid.
+        """
+        return self.partition.grid
+
+    @property
+    def det_radius(self):
+        """Detector circle radius of this geometry."""
+        return self._det_radius
+
+    def det_refpoint(self, angle):
+        """The detector reference point function.
+        Parameters
+        ----------
+        angle : `float`
+            The parameter angle `float` given in radian. It must be
+            contained in this geometry's motion parameter set.
+        Returns
+        -------
+        point : `numpy.ndarray`, shape (3,)
+            The reference point on a circle in the transaxial plane with
+            radius ``R`` and on at a axial position ``z``.
+            Transaxial point given by the rotation angle ``phi``.
+        """
+        origin_to_det = self.det_radius * self._center_to_det
+        circle_component = self.rotation_matrix(angle).dot(origin_to_det)
+
+        return circle_component
```

Comment:
Having read this, I don't see a substantial difference to `Parallel3dAxisGeometry` except that there is a radius for the detector. Perhaps easiest is to just subclass that one and add the radius?